x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/controller.go

```diff
@@ -105,25 +121,41 @@ func (e *Controller) Update(c *config.Config) error {
 	e.ast = rawAst
 	e.lock.Unlock()
 
-	return e.update()
+	rErr = e.update(ctx)
+	return
 }
 
 // Set sets the transpiler vars for dynamic inputs resolution.
-func (e *Controller) Set(vars []*transpiler.Vars) {
+func (e *Controller) Set(ctx context.Context, vars []*transpiler.Vars) {
+	var err error
+	span, ctx := apm.StartSpan(ctx, "Set", "app.internal")
+	defer func() {
+		if err != nil {
+			apm.CaptureError(ctx, err).Send()
+		}
```

Comment:
[Suggestion]
It's safe to call `apm.CaptureError` with a nil error and then call `Send`. Therefore it can be simplified.
```suggestion
		apm.CaptureError(ctx, err).Send()
```